cpp/src/io/orc/stripe_data.cu

```diff
@@ -1728,9 +1728,9 @@ CUDF_KERNEL void __launch_bounds__(block_size)
         // Adjust the maximum number of values
         if (numvals == 0 && vals_skipped == 0) {
           numvals = s->top.data.max_vals;  // Just so that we don't hang if the stream is corrupted
-        } else {
-          if (t == 0 && numvals < s->top.data.max_vals) { s->top.data.max_vals = numvals; }
         }
+        __syncthreads();
+        if (t == 0) { s->top.data.max_vals = cuda::std::min(s->top.data.max_vals, numvals); }
```

Comment:
Another reported error was about `s->top.data.max_vals` being read in the condition and written in the block. I don't know why this is a race, but switching to `min` fixes it. I also prefer using `min` here, so no harm IMO.